crates/bevy_pbr/src/wireframe.rs

```diff
@@ -53,31 +72,179 @@ fn extract_wireframe_config(mut commands: Commands, wireframe_config: Res<Wirefr
+        commands.get_or_spawn(entity).insert(wireframe.clone());
+    }
+}
+
+fn prepare_wireframes(
+    mut commands: Commands,
+    config: Res<WireframeConfig>,
+    render_device: Res<RenderDevice>,
+    render_queue: Res<RenderQueue>,
+    mut wireframe_meta: ResMut<GlobalWireframeMeta>,
+    global_query: Query<(Entity, Option<&Wireframe>), (With<Handle<Mesh>>, With<MeshUniform>)>,
+    wireframe_query: Query<
+        (Entity, Option<&Wireframe>),
+        (With<Handle<Mesh>>, With<MeshUniform>, With<Wireframe>),
+    >,
+) {
+    wireframe_meta.uniforms.clear();
+    wireframe_meta.uniforms.push(WireframeUniform {
+        color: config.default_color.as_linear_rgba_f32().into(),
+    });
+
+    let add_wireframe_uniform = |(entity, wireframe): (Entity, Option<&Wireframe>)| {
+        let custom_color = wireframe.map(|wireframe| wireframe.color);
+        let uniform_offset = WireframeUniformOffset(if let Some(custom_color) = custom_color {
+            wireframe_meta.uniforms.push(WireframeUniform {
+                color: custom_color.as_linear_rgba_f32().into(),
+            })
+        } else {
+            0
+        });
+        commands.entity(entity).insert(uniform_offset);
+    };
+
+    if config.global {
+        global_query.for_each(add_wireframe_uniform);
+    } else {
+        wireframe_query.for_each(add_wireframe_uniform);
+    }
+
+    wireframe_meta
+        .uniforms
+        .write_buffer(&render_device, &render_queue);
+}
+
+/// Internal [`WireframePlugin`] resource.  
+/// [`GlobalWireframeBindGroup`] stores the [`BindGroup`] of wireframe data that is used on the GPU side.
+struct GlobalWireframeBindGroup {
+    bind_group: BindGroup,
+}
+
+fn queue_wireframes_bind_group(
+    mut commands: Commands,
+    render_device: Res<RenderDevice>,
+    meta: Res<GlobalWireframeMeta>,
+    bind_group: Option<Res<GlobalWireframeBindGroup>>,
+) {
+    if bind_group.is_none() {
+        commands.insert_resource(GlobalWireframeBindGroup {
+            bind_group: render_device.create_bind_group(&BindGroupDescriptor {
+                entries: &[BindGroupEntry {
+                    binding: 0,
+                    resource: meta.uniforms.binding().unwrap(),
+                }],
+                label: Some("wireframe_bind_group"),
+                layout: &meta.bind_group_layout,
+            }),
+        });
     }
 }
 
-/// Controls whether an entity should rendered in wireframe-mode if the [`WireframePlugin`] is enabled
-#[derive(Component, Debug, Clone, Default, Reflect)]
+/// Controls per-entity wireframe rendering settings (if the [`WireframePlugin`] is enabled)
+#[derive(Component, Debug, Clone, Reflect)]
 #[reflect(Component)]
-pub struct Wireframe;
+pub struct Wireframe {
+    /// The color of this wireframe.  
+    /// For the [`Entity`] that has this [`Wireframe`] component, this color overrides any that was set in [`WireframeConfig::default_color`].
+    pub color: Color,
```

Comment:
If we made this an `Option<Color>`, we could derive `Default` nicely here, and solve the config-coupling issue I mention below.